docs/resources/data-representation/data-types.md

```diff
@@ -0,0 +1,60 @@
+---
+slug: data-types
+title: Data Types
+tags: []
+description: Dive into the diverse range of data types supported by FlutterFlow, from basic primitives like integers and strings to complex composite types and built-in functionalities tailored for app development.
+sidebar_position: 1
+---
+
+# Data Types
+
+FlutterFlow supports a variety of data types to accommodate different needs in your app. These data types range from the basic, such as integers and strings, to more complex types like lists, maps, and built-in data types.
+
+## Primitive Data Types
+
+Primitive data types are the most basic data types. They include integers, doubles, booleans, and strings. These are the building blocks and are essential in any kind of app development.
+
+## Composite Data Types
+
+Composite data types are made up of primitive data types. They can hold multiple values and can be used to structure and organize data in a more meaningful way. Examples of composite data types include lists, such as a list of fruit names.
+
+## Built-in Data Types
+
+The built-in data types cater to different aspects of app development. Each data type is specifically created to handle common app development requirements, such as media handling, data storage, location services, and time tracking. They are part of what makes FlutterFlow attractive for developers looking to speed up app development.
+
+Below is a list of all supported built-in data types:
+
+- **Color**: Stores color values.
+- **Image Path**: Stores the URL of uploaded images.
+- **Video Path**: Stores the URL of uploaded videos.
+- **Audio Path**: Stores the URL of uploaded audio files.
+- **Document Reference**: Stores references to documents, simplifying data fetching.
+- **Document**: Stores actual Firestore documents.
+- **Timestamp**: Stores date and time values.
+- **Json**: Stores JSON values, such as `{"firstName":"John", "lastName":"Doe"}`.
+- **LatLng**: Stores the latitude and longitude of specific locations, aiding Google Maps integration.
+- **TimestampRange**: Stores start and end date-time values.
+- **GooglePlace**: Stores GooglePlace data.
+- **Data Type**: Stores custom data types.
+- **Supabase Row**: Stores actual row data from a Supabase table.
+- **Uploaded File (Bytes)**: Stores uploaded files in Bytes.
+
+## Custom Data Types
+
+You can also create your own custom data types. This can be especially useful when you need a specific structure for your data that doesn't fit into the predefined types. For example, you might create a custom data type for a user profile, which includes several pieces of data like a name, an email address, and a profile picture.
```

Comment:
Maybe showing a visual with example would be good. We can show the different fields of the custom data type, like a class diagram but more non-developer friendly diagram/

Reply:
Ok, created a visual and added it inside the dedicated page of custom data type. Not adding it here because this page gives an overview of all the data types.